### PR TITLE
feat: add `static-color` to replace `static`

### PR DESCRIPTION
**Pull request by rubencarvalho**
## Description
This PR deprecates the static attribute and introduces the new staticColor attribute.

## Related issue(s)

- [SWC-512](https://jira.corp.adobe.com/browse/SWC-512)

## Motivation and context
 The `staticColor` attribute more clearly conveys the attribute’s purpose, addressing confusion caused by the reserved `static` keyword in JavaScript/TypeScript and aligning better with the Spectrum design system documentation.

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/action-button/test/action-button.test.ts [hunk 1]
```diff
@@ -343,5 +317,41 @@ describe('ActionButton', () => {
...
+            const spyCall = consoleWarnStub.getCall(0);
+            expect(
+                (spyCall.args.at(0) as string).includes('"static"'),
+                'confirm static-centric message'
+            ).to.be.true;
+            expect(spyCall.args.at(-1), 'confirm `data` shape').to.deep.equal({
+                data: {
+                    localName: 'sp-action-button',
+                    type: 'api',
+                    level: 'default',
```
Comment (nikkimk) [inline packages/action-button/test/action-button.test.ts:338]: @rubencarvalho Should level be 'default'?

**Review by nikkimk — DISMISSED** [review]
Looks good to me.

File: packages/action-button/test/action-button.test.ts (hunk 1 above)
Comment (rubencarvalho) [reply packages/action-button/test/action-button.test.ts:338]: Oops, nice catch.

**Review by blunteshwar — APPROVED** [review]
LGTM!

**Pull request merged**